### DISCUSSION TITLE
How to download submodules (nidr)?

### DISCUSSION THREAD

**Discussion by erdchris** [snl-dakota/dakota#72] [Q&A] [answered]
Sorry, never built Dakota before. This is for a RHEL 8 box. I've downloaded the tar ball dakota-6.18.0.tar.gz but after following the web guide and INSTALL file, it fails immediately. 

```
> cmake -C ../cmake/BuildDakotaCustom.cmake ..
loading initial cache file ../cmake/BuildDakotaCustom.cmake
-- Configuring Dakota with CMake 3.21.0
CMake Error at CMakeLists.txt:33 (message):
  Required source sub-directory
  dakota-6.18/packages/external/nidr does not
  exist; if using a Git repository, are the submodules checked out?
```

**Comment by jadamstephens** [comment]
How did you download the source?

[This source download](https://github.com/snl-dakota/dakota/releases/download/v6.18.0/dakota-6.18.0-public-src-cli.tar.gz) has the packages/external/nidr folder and should work.

Adam

**Reply by erdchris** [reply]
Ok, thank you. I just went to "Source Code (tar.gz)" for https://github.com/snl-dakota/dakota/archive/refs/tags/v6.18.0.tar.gz

**Reply by jadamstephens** [reply, marked as answer]
That package is generated automatically by GitHub. I wish we could disable it, because as you found, it doesn't work. Dakota has a number of git submodules that GitHub doesn't properly pull in when it generates those packages.
